VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java

```diff
@@ -272,33 +272,125 @@ private void startOnlineDownload(DownloadEntry videoData, ProgressWheel progress
     }
 
 
+    public synchronized void playVideoModel(final DownloadEntry video) {
+        try {
+            if (playerFragment.isPlaying()) {
+                if (video.getVideoId().equals(playerFragment.getPlayingVideo().getVideoId())) {
+                    logger.debug("this video is already being played, skipping play event");
+                    return;
+                }
+            }
+        } catch(Exception ex) {}
+
+
+        try{
+
+            // reload this model
+            new Storage(getActivity()).reloadDownloadEntry(video);
```

Comment:
@aleffert  changed to
TranscriptModel transcript = getTranscriptModel(video);
 String filepath = getVideoPath(video);
 updateLastAccess(video);